### PR TITLE
Better error message on bogus constructor param to elements and other model items

### LINKED ISSUE #50: Make error message more intuitive if you get a wrong parameter on an element constructor
#### Problem description

It's easy to get a constructor argument wrong, but at the moment the error you get back isn't very helpful:
```python
>>> Container(name="Container", bar=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  ...
  File "c:\...\structurizr-python\src\structurizr\model\model_item.py", line 86, in __init__
    super().__init__(**kwargs)
TypeError: object.__init__() takes exactly one argument (the instance to initialize)
```

This should be more in line with a regular object:
```python
>>> class Foo:
...   def __init__(self):
...     pass
...
>>> Foo(bar=7)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'bar'
```

#### Code Sample

Create a [minimal, complete, verifiable example](https://stackoverflow.com/help/mcve).

```python
from structurizr.model import Container
Container(name="container", foobar=1)
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\...\structurizr-python\src\structurizr\model\container.py", line 98, in __init__
    super().__init__(**kwargs)
  File "c:\...\structurizr-python\src\structurizr\model\element.py", line 71, in __init__
    super().__init__(**kwargs)
  File "c:\...\structurizr-python\src\structurizr\mixin\model_ref_mixin.py", line 35, in __init__
    super().__init__(**kwargs)
  File "c:\...\structurizr-python\src\structurizr\model\model_item.py", line 86, in __init__
    super().__init__(**kwargs)
TypeError: object.__init__() takes exactly one argument (the instance to initialize)
```

### Context

<details>

```
System Information
==================
OS         Windows
OS-release      10
Python       3.7.5

Package Versions
================
depinfo                              1.5.4
httpx                               0.16.1
importlib_metadata                   1.7.0
ordered-set                            3.1
pip                                 20.2.4
pydantic                             1.7.1
python-dotenv                       0.14.0
setuptools                          41.2.0
structurizr-python 0.1.1+32.g8fa91ab.dirty
```

</details>

### PR DESCRIPTION
**Pull request by yt-ms**
* [X] fix https://github.com/Midnighter/structurizr-python/issues/50
* [X] description of feature/fix
* [X] tests added/passed
* [X] add an entry to the [next release](../CHANGELOG.rst)

Added check in `ModelItem` for any unused parameters, and now raises meaningful `TypeError` listing them.

------------------------------

THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE Apache License v.2.0. YOU MAY OBTAIN A COPY OF THE LICENSE AT https://www.apache.org/licenses/LICENSE-2.0.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS

### CODE REVIEW

**Review by Midnighter — CHANGES_REQUESTED** [review]
I agree with the idea but it's slightly trickier than your current solution. The reason I tend to pass keyword arguments into the `super`'s constructor is to enable multiple inheritance.

If you decide against multiple inheritance, an easy solution is to not pass on keyword arguments in classes who inherit from `object` and silently ignore extra arguments. (As you now basically do with `super().__init__()` just that you inspect `kwargs` first.)

However, in multiple inheritance, for example, class `C` inherits from both classes `P1` and `P2` which inherit from `object`, you don't know in which order the MRO will turn out. Thus `super()` of `P1` may be `object` or `P2` and the same is true for `P2`. Thus you need to pass on keyword arguments in order to properly initialize all of them. That means, extra arguments not consumed by `P1` and `P2` will be passed on to `object` which will then complain.

So in summary, we should still use `super().__init__(**kwargs)` but we can inspect `type(self).__mro__` and if `object` is next in line, we can raise a `TypeError` as you suggest.

**Comment by Midnighter** [comment]
Closing this one due to #60.
